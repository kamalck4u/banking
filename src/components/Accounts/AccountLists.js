import React from 'react';
import { useHistory } from 'react-router-dom';
import { ACCOUNT_NUMBER, ACCOUNT_NAME, ACCOUNT_TYPE, CURRENCY, ACCOUNT_BALANCE, LOADER_TEXT_ACCOUNTS } from '../Common/Constants';
import '../../App.css';
import { Loader } from '../Common/Panels';
import { array, bool } from 'prop-types';

const AccountLists = ({ accounts, isLoading }) => {
	const history = useHistory();
	return (
		<React.Fragment>
			{
				isLoading ? <Loader loaderText={LOADER_TEXT_ACCOUNTS} />
					:
					<div className="table-responsive">
						<table className="table table-striped">
							<thead className="thead-dark">
								<tr>
									<th scope="col">{ACCOUNT_NUMBER}</th>
									<th scope="col">{ACCOUNT_NAME}</th>
									<th scope="col">{ACCOUNT_TYPE}</th>
									<th scope="col">{CURRENCY}</th>
									<th scope="col">{ACCOUNT_BALANCE}</th>
								</tr>
							</thead>
							<tbody className="row-clickable">
								{accounts.map((account) => {
									return (
										<tr key={account.id} className="clickable-row table-cell" onClick={() => history.push(`/accounts/${account.id}/transactions`, { account })}>
											<td>{account.account_number}</td>
											<td>{account.account_name}</td>
											<td>{account.account_type}</td>
											<td>{account.currency}</td>
											<td>{account.balance}</td>
										</tr>
									);
								}
								)}
							</tbody>
						</table>
					</div>
			}
		</React.Fragment>
	);
};

AccountLists.propTypes = {
	accounts: array,
	isLoading: bool
}

export default AccountLists;
