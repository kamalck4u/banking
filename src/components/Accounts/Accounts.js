import React, { useState } from 'react';
import { FetchAccountLists } from '../../apis/FetchAccountDetails';
import AccountLists from './AccountLists';
import { HeaderPanel, GenericErrorPanel } from '../Common/Panels';
import { WELCOME_TEXT, TAG_LINE } from '../Common/Constants';

const Accounts = () => {
	const [accounts, setAccounts] = useState([]);
	const [error, setError] = useState({});
	const [isLoading, setIsLoading] = useState();

	React.useEffect(() => {
		setIsLoading(true);
		(async () => {
			try {
				const response = await FetchAccountLists();
				setAccounts(response.data);
			} catch (error) {
				setError({ isFailed: true, error: error });
			}
			setIsLoading(false);
		}
		)();
	}, []);

	return (
		<React.Fragment>
			<HeaderPanel header={WELCOME_TEXT} content={TAG_LINE} />
			{
				error.isFailed
					? <GenericErrorPanel />
					: <AccountLists accounts={accounts} isLoading={isLoading} />
			}
		</React.Fragment>
	);
};

export default Accounts;
