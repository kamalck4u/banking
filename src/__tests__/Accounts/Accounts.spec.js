import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Accounts from '../../components/Accounts/Accounts';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Accounts', () => {
	it('snapshot renders', () => {
		const Accountscomponent = renderer.create(<Accounts />);
		expect(Accountscomponent.toJSON()).toMatchSnapshot();
	});
});

describe('Accounts Home page', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(<Accounts />);
	});

	it('should display welcome message', () => {
		expect(wrapper.find('HeaderPanel').props()).toEqual({ content: 'Building the Future of Trade Finance', header: 'Welcome CONTOUR' });
	});

	it('should render AccountLists component', () => {
		expect(wrapper.find('AccountLists')).toHaveLength(1);
	});

	it('should not render GenericErrorPanel component', () => {
		expect(wrapper.find('GenericErrorPanel')).toHaveLength(0);
	});
});

