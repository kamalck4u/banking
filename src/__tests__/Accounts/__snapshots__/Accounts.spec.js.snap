// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accounts snapshot renders 1`] = `
Array [
  <div
    className="jumbotron panel-header"
  >
    <h1
      className="display-4"
    >
      Welcome CONTOUR
    </h1>
    <p
      className="lead"
    >
      Building the Future of Trade Finance
    </p>
    <hr
      className="my-4"
    />
    <p>
       
    </p>
  </div>,
  <div
    className="table-responsive"
  >
    <table
      className="table table-striped"
    >
      <thead
        className="thead-dark"
      >
        <tr>
          <th
            scope="col"
          >
            Account Number
          </th>
          <th
            scope="col"
          >
            Account Name
          </th>
          <th
            scope="col"
          >
            Account Type
          </th>
          <th
            scope="col"
          >
            Currency
          </th>
          <th
            scope="col"
          >
            Balance
          </th>
        </tr>
      </thead>
      <tbody
        className="row-clickable"
      />
    </table>
  </div>,
]
`;
